easybuild/easyconfigs/r/R/R-4.1.0-foss-2021a.eb

```diff
@@ -0,0 +1,2921 @@
+name = 'R'
+version = '4.1.0'
+
+homepage = 'https://www.r-project.org/'
+description = """R is a free software environment for statistical computing
+ and graphics."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+source_urls = ['https://cloud.r-project.org/src/base/R-%(version_major)s']
+sources = [SOURCE_TAR_GZ]
+patches = ['%(name)s-%(version)s_identify-flexiblas-in-configure.patch']
+checksums = [
+    'e8e68959d7282ca147360fc9644ada9bd161bab781bab14d33b8999a95182781',  # R-4.1.0.tar.gz
+    '2c6720e2e144ae4fe00842daab0ebba72241080603e0ff1a6ca758738041b257',  # R-4.1.0_identify-flexiblas-in-configure.patch
+]
+
+builddependencies = [
+    ('pkg-config', '0.29.2'),
+    ('Xvfb', '1.20.11'),
+]
+dependencies = [
+    ('X11', '20210518'),
+    ('Mesa', '21.1.1'),
+    ('libGLU', '9.0.1'),
+    ('cairo', '1.16.0'),
+    ('libreadline', '8.1'),
+    ('ncurses', '6.2'),
+    ('bzip2', '1.0.8'),
+    ('XZ', '5.2.5'),
+    ('zlib', '1.2.11'),
+    ('SQLite', '3.35.4'),
+    ('PCRE2', '10.36'),
+    ('libpng', '1.6.37'),  # for plotting in R
+    ('libjpeg-turbo', '2.0.6'),  # for plottting in R
+    ('LibTIFF', '4.2.0'),
+    ('Java', '11', '', True),
+    ('Tk', '8.6.11'),  # for tcltk
+    ('cURL', '7.76.0'),  # for RCurl
+    ('libxml2', '2.9.10'),  # for XML
+    ('GMP', '6.2.1'),  # for igraph
+    ('NLopt', '2.7.0'),  # for nloptr
+    ('FFTW', '3.3.9'),  # for fftw
+    ('libsndfile', '1.0.31'),  # for seewave
+    ('ICU', '69.1'),  # for rJava & gdsfmt
+    ('HDF5', '1.10.7'),  # for hdf5r
+    ('UDUNITS', '2.2.28'),  # for units
+    ('GSL', '2.7'),  # for RcppGSL
+    ('ImageMagick', '7.0.11-14'),  # for animation
+    ('GLPK', '5.0'),  # for Rglpk
+    ('nodejs', '14.17.0'),  # for V8 (required by rstan)
+    ('GDAL', '3.3.0'),  # for sf
+    ('OpenSSL', '1.1', '', True),
+]
+
+preconfigopts = 'export FLIBS="$FLIBS -lm" && '
```

Comment:
At the very least a comment should be added above to explain why this is needed, but it should be possible to come up with something specific to `mclust`...

Reply:
I've removed this line and gone with the patch option, as per https://github.com/easybuilders/easybuild-easyconfigs/pull/13032#issuecomment-855378649 - done in cd4cd5e and c0a7dea

Reply:
Scratch that. We hit a similar error on `quantreg` and `waveslim`. I'd expect it for other extensions as well. I've reverted and added a comment in 88c0e22.

Reply:
@branfosj I tried installing this easyconfig without the top-level `preconfigopts`, and it works fine for me...

Can you share the error output you're getting for one of these extensions, along with the failing command, so I can compare?

On which OS are you seeing trouble?

Reply:
RHEL 8.3

Error is: `/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs/mclust.so: undefined symbol: _ZGVbN2v_log`

Build for `mclust` is:
```
== 2021-06-03 19:08:22,248 filetools.py:1775 INFO Creating directory /dev/shm/build-branfosj-admin/branfosj-admin-up/R/4.1.0/foss-2021a/mclust (parents: True, set_gid: False, sticky: False)
== 2021-06-03 19:08:22,248 run.py:226 INFO running cmd: tar xzf /rds/bear-sysadmin/configmgmt/easybuild/sources/r/R/extensions/mclust_5.4.7.tar.gz 
== 2021-06-03 19:08:22,349 run.py:226 INFO running cmd:  R CMD INSTALL /dev/shm/build-branfosj-admin/branfosj-admin-up/R/4.1.0/foss-2021a/mclust/mclust   --library=/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library --no-clean-on-error  
== 2021-06-03 19:08:45,924 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): cmd " R CMD INSTALL /dev/shm/build-branfosj-admin/branfosj-admin-up/R/4.1.0/foss-2021a/mclust/mclust   --library=/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library --no-clean-on-error " exited with exit code 1 and output:
* installing *source* package mclust ...
** package mclust successfully unpacked and MD5 sums checked
** using staged installation
** libs
gfortran -fno-optimize-sibling-calls  -fpic  -O2 -ftree-vectorize -march=native -fno-math-errno  -c dmvnorm.f -o dmvnorm.o
gcc -I"/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/include" -DNDEBUG   -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenSSL/1.1/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GDAL/3.3.0-foss-2021a/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/nodejs/14.17.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GLPK/5.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ImageMagick/7.0.11-14-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GSL/2.7-GCC-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/UDUNITS/2.2.28-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/HDF5/1.10.7-gompi-2021a/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ICU/69.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libsndfile/1.0.31-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FFTW/3.3.9-gompi-2021a/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/NLopt/2.7.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GMP/6.2.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libxml2/2.9.10-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cURL/7.76.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Tk/8.6.11-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Java/11.0.2/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/LibTIFF/4.2.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libjpeg-turbo/2.0.6-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libpng/1.6.37-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/PCRE2/10.36-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/SQLite/3.35.4-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/zlib/1.2.11-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/XZ/5.2.5-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/bzip2/1.0.8-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ncurses/6.2-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libreadline/8.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cairo/1.16.0-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libGLU/9.0.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Mesa/21.1.1-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/X11/20210518-GCCcore-10.3.0/include -I/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FlexiBLAS/3.0.4-GCC-10.3.0/include   -fpic  -O2 -ftree-vectorize -march=native -fno-math-errno  -c init.c -o init.o
gfortran -fno-optimize-sibling-calls  -fpic  -O2 -ftree-vectorize -march=native -fno-math-errno  -c mclust.f -o mclust.o
gfortran -fno-optimize-sibling-calls  -fpic  -O2 -ftree-vectorize -march=native -fno-math-errno  -c mclustaddson.f -o mclustaddson.o
gcc -shared -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenSSL/1.1/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/OpenSSL/1.1/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GDAL/3.3.0-foss-2021a/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GDAL/3.3.0-foss-2021a/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/nodejs/14.17.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/nodejs/14.17.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GLPK/5.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GLPK/5.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ImageMagick/7.0.11-14-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ImageMagick/7.0.11-14-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GSL/2.7-GCC-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GSL/2.7-GCC-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/UDUNITS/2.2.28-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/UDUNITS/2.2.28-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/HDF5/1.10.7-gompi-2021a/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/HDF5/1.10.7-gompi-2021a/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ICU/69.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ICU/69.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libsndfile/1.0.31-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libsndfile/1.0.31-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FFTW/3.3.9-gompi-2021a/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FFTW/3.3.9-gompi-2021a/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/NLopt/2.7.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/NLopt/2.7.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GMP/6.2.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GMP/6.2.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libxml2/2.9.10-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libxml2/2.9.10-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cURL/7.76.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cURL/7.76.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Tk/8.6.11-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Tk/8.6.11-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Java/11.0.2/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Java/11.0.2/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/LibTIFF/4.2.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/LibTIFF/4.2.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libjpeg-turbo/2.0.6-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libjpeg-turbo/2.0.6-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libpng/1.6.37-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libpng/1.6.37-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/PCRE2/10.36-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/PCRE2/10.36-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/SQLite/3.35.4-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/SQLite/3.35.4-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/zlib/1.2.11-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/zlib/1.2.11-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/XZ/5.2.5-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/XZ/5.2.5-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/bzip2/1.0.8-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/bzip2/1.0.8-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ncurses/6.2-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ncurses/6.2-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libreadline/8.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libreadline/8.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cairo/1.16.0-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/cairo/1.16.0-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libGLU/9.0.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/libGLU/9.0.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Mesa/21.1.1-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Mesa/21.1.1-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/X11/20210518-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/X11/20210518-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Xvfb/1.20.11-GCCcore-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Xvfb/1.20.11-GCCcore-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ScaLAPACK/2.1.0-gompi-2021a-fb/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/ScaLAPACK/2.1.0-gompi-2021a-fb/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FlexiBLAS/3.0.4-GCC-10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/FlexiBLAS/3.0.4-GCC-10.3.0/lib -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GCCcore/10.3.0/lib64 -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GCCcore/10.3.0/lib -o mclust.so dmvnorm.o init.o mclust.o mclustaddson.o -lflexiblas -lgfortran -lgfortran -lgfortran -L/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/lib -lR
installing to /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for mclust in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs/mclust.so':
  /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/R/4.1.0-foss-2021a/lib64/R/library/00LOCK-mclust/00new/mclust/libs/mclust.so: undefined symbol: _ZGVbN2v_log
Error: loading failed
Execution halted
ERROR: loading failed
 (at easybuild/tools/run.py:565 in parse_cmd_output)
```

Reply:
Linking command looks identical on CentOS 7, no problem there. I'll also check on RHEL8

Reply:
I'm also seeing this on our Ubuntu 20.04 (singularity container install).

There are other options for placing the `-lm` - I can add it to `LDFLAGS` instead and it still works. However, it does look like it needs setting in the R ec for me, as these items get stored in a config file inside the R install.

Reply:
I can confirm I see the same problem on RHEL8, the `-lm` needs to be specified explicitly.

I'm not sure whether it makes more sense to add it through `LDFLAGS`...

Reply:
The contents of `<installdir>/lib/R/etc/Makeconf` looks identical on both CentOS 7 and RHEL8 (except for absolute paths).

```
FLIBS = -lgfortran
```

Is there in both, yet somehow `-lm` is added automagically, or not needed at all, when linking `mclust` on CentOS 7?!

Reply:
CentOS 7 brings in further differences. The missing symbols (`_ZGVbN2v_log`) are part of the vectorised math library (libmvec) so something different could be happening there. It confuses me that I've not seen this before.